docs/get-started/git/git-basics.md

```diff
@@ -0,0 +1,95 @@
+drwxr-xr-x+ 99 ceford  staff  3168 26 Aug 06:31 ..
+drwxr-xr-x   9 ceford  staff   288 26 Aug 07:02 .git
+-rw-r--r--   1 ceford  staff  1689 26 Aug 07:10 git.md
+```
+
+You can look at the contents of the .git folder if you wish but as a rule you should not make any changes there yourself. Let the git commands do the work. You can delete the .git folder if you wish without affecting the source text but you will lose any history of changes.
+
+### Staging
+
+The next step is to issue a command to add a source file to the index of files that git keeps track of:
+
+```
+git add git.md
+```
+
+There is no response to that command - git just does it.
+
+### Commit
+
+The commit command tells git to record all changes made to the source files so far:
+
+```
+git commit -m "Text entered as far as Commit"
+```
+
+To which the response is:
+
+```
+[master (root-commit) 913815c] Text entered as far as Commit
+ 1 file changed, 45 insertions(+)
+ create mode 100644 git.md
+```
+
+That strange number, 913815c, is the commit **id** that allows git to keep track of that particular point in history. The text in quotes is the commit message - it should be a short informative description of the changes made in that commit.
+
+### Branch
+
+Suppose you want to try a new feature but you are not sure whether you will use it. Or, you are working on a collaborative project and you want to change something contributed by others. This is where branching comes in. You make a branch of your repo, checkout the branch, make your changes and then decide what to do. Try this:
+
+```
+git branch --list
+* master
+```
+
+The list command indicates that this repo has only one branch, named master. Try these commands:
+
+```
+git branch myfix
+git checkout myfix
+git branch --list 
+  master
+* myfix
+```
+The list shows that there are now two branches and the * indicates that the myfix branch is checked out as the current branch. I can now go ahead and make changes to the myfix branch without affecting the existing content of the master branch. If I decide to use the changes I need to commit them and then either merge the myfix branch into my master branch or ask my collaborator to merge my myfix branch into his master branch.
+
+When the branch is finished with, by being merged or abandoned, it can be deleted.
+
+## Full Documentation
+
+Git has many more commands and options. Mostly you don't need to know them because your IDE handles them for you with Menu items. If you want to look up what a menu item does try this source:
+
+[Git Documentation](https://git-scm.com/doc)
+
+If you have been experimenting, remember you can delete the repo (.git) and you will be left with your source files as you see them in your editor.
+
+## Using an IDE
+
+One final point: your chosen IDE will not put any IDE specific content in your git repo. It follows that you can change your IDE and expect it to work equally well with your existing repo. 
```

Comment:
that depends on the ide or desktop git application. smartgit for example will store a config file and a cache in the .git folder. However your comment that you can change tools without any impact is still correct.